tests/functional/u/unnecessary/unnecessary_comprehension.py

```diff
@@ -34,3 +37,13 @@
 {y: z for x in iterable for y, z in x}  # exclude nested comprehensions
 {x: 1 for x in iterable}  # expression != target_list
 {2 * x: 3 + x for x in iterable}  # exclude useful comprehensions
+
+# Some additional tests on helptext -- when object is already a list/set/dict
+my_list = list()
+my_dict = dict()
+my_set = set()
+
+[elem for elem in my_list]  # [unnecessary-comprehension]
+{k: v for k, v in my_dict.items()} # [unnecessary-comprehension]
+{k: my_dict[k] for k in my_dict} # [consider-using-dict-items]
```

Comment:
I decided not to include the logic to emit ``unnecessary-comprehension`` here, as doing so is not so trivial.

Furthermore, ``consider-using-dict-items`` would point users towards writing ```{k: v for k, v in my_dict.items()}```, which will then emit ``unnecessary-comprehension`` @Pierre-Sassoulas do you think that is okay?

Reply:
Yes let's not overcomplicate, there is also the time to compute to take into account better a fast good enough check than a perfect slow one :)